Fedora based image is broken

Even thought #262 fixed #261 test for Fedora based image seem to fail in general:

```
make[1]: Entering directory '/home/goern/Source/sclorg/s2i-python-container'
VERSION="3.6" SKIP_SQUASH=1 UPDATE_BASE= OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=. OPENSHIFT_NAMESPACES="3.3" CUSTOM_REPO="" /usr/bin/env bash common/build.sh
-> Version 3.6: building image from 'Dockerfile.fedora' ...
Sending build context to Docker daemon 58.88 kB
Step 1/13 : FROM registry.fedoraproject.org/f27/s2i-base:latest
 ---> 07f207eb54b6
Step 2/13 : EXPOSE 8080
 ---> Using cache
 ---> c36365c6fc84
Step 3/13 : ENV PYTHON_VERSION 3.6 PATH $HOME/.local/bin/:$PATH PYTHONUNBUFFERED 1 PYTHONIOENCODING UTF-8 LC_ALL en_US.UTF-8 LANG en_US.UTF-8 PIP_NO_CACHE_DIR off
 ---> Using cache
 ---> 9b978883e980
Step 4/13 : ENV NAME python3 VERSION 0 RELEASE 1 ARCH x86_64
 ---> Using cache
 ---> e851753a1741
Step 5/13 : ENV SUMMARY "Platform for building and running Python $PYTHON_VERSION applications" DESCRIPTION "Python $PYTHON_VERSION available as container is a base platform for building and running various Python $PYTHON_VERSION applications and frameworks. Python is an easy to learn, powerful programming language. It has efficient high-level data structures and a simple but effective approach to object-oriented programming. Python's elegant syntax and dynamic typing, together with its interpreted nature, make it an ideal language for scripting and rapid application development in many areas on most platforms."
 ---> Using cache
 ---> 6c7874b34c98
Step 6/13 : LABEL summary "$SUMMARY" description "$DESCRIPTION" io.k8s.description "$DESCRIPTION" io.k8s.display-name "Python 3.6" io.openshift.expose-services "8080:http" io.openshift.tags "builder,python,python36,rh-python36" com.redhat.component "$NAME" name "$FGC/$NAME" version "$VERSION" release "$RELEASE.$DISTTAG" architecture "$ARCH" usage "s2i build https://github.com/sclorg/s2i-python-container.git --context-dir=3.6/test/setup-test-app/ $FGC/$NAME python-sample-app" maintainer "SoftwareCollections.org <sclorg@redhat.com>"
 ---> Using cache
 ---> caa19172a75f
Step 7/13 : RUN INSTALL_PKGS="python3 python3-devel python3-setuptools python3-pip python3-virtualenv 	 nss_wrapper httpd httpd-devel atlas-devel gcc-gfortran libffi-devel libtool-ltdl 	 enchant" &&     dnf install -y --setopt=tsflags=nodocs $INSTALL_PKGS &&     rpm -V $INSTALL_PKGS &&     dnf clean all -y
 ---> Using cache
 ---> 76b07912d189
Step 8/13 : COPY ./s2i/bin/ $STI_SCRIPTS_PATH
 ---> Using cache
 ---> 5ee41b6ffe2d
Step 9/13 : COPY ./root/ /
 ---> Using cache
 ---> b27fabc084b0
Step 10/13 : RUN virtualenv-$PYTHON_VERSION ${APP_ROOT} &&     chown -R 1001:0 ${APP_ROOT} &&     fix-permissions ${APP_ROOT} -P
 ---> Using cache
 ---> d14d9fd583c8
Step 11/13 : USER 1001
 ---> Using cache
 ---> 8698eca708e9
Step 12/13 : CMD $STI_SCRIPTS_PATH/usage
 ---> Using cache
 ---> 7feacc58f065
Step 13/13 : LABEL "io.openshift.builder-version" '"0717c35"'
 ---> Using cache
 ---> 608a84f5032e
Successfully built 608a84f5032e
python 3.6 => 608a84f5032e
make[1]: Leaving directory '/home/goern/Source/sclorg/s2i-python-container'
VERSIONS="3.6" SKIP_SQUASH=1 UPDATE_BASE= OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=. OPENSHIFT_NAMESPACES="3.3" CUSTOM_REPO="" TEST_MODE=true /usr/bin/env bash common/tag.sh
-> Tagging image '608a84f5032e' as 'f27/python3:0' and 'f27/python3:latest'
VERSIONS="3.6" SKIP_SQUASH=1 UPDATE_BASE= OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=. OPENSHIFT_NAMESPACES="3.3" CUSTOM_REPO="" TEST_MODE=true /usr/bin/env bash common/test.sh
test/run: line 12: warning: command substitution: ignored null byte in input
test/run: line 13: warning: command substitution: ignored null byte in input

[INFO] Testing 's2i usage' ...


[INFO] Testing 'docker run' usage ...


[INFO] Preparing to test standalone-test-app ...

Reinitialized existing Git repository in /home/goern/Source/sclorg/s2i-python-container/examples/standalone-test-app/.git/
On branch master
nothing to commit, working tree clean

[INFO] Building the standalone-test-app application image ...

I0407 12:38:18.962971 30068 util.go:189] Checking if image "f27/python3" is available locally ...
I0407 12:38:18.972333 30068 clone.go:32] Downloading "/home/goern/Source/sclorg/s2i-python-container/3.6/test/standalone-test-app" ...
I0407 12:38:18.991282 30068 install.go:251] Using "assemble" installed from "image:///usr/libexec/s2i/assemble"
I0407 12:38:18.991340 30068 install.go:251] Using "run" installed from "image:///usr/libexec/s2i/run"
I0407 12:38:18.991358 30068 install.go:251] Using "save-artifacts" installed from "image:///usr/libexec/s2i/save-artifacts"
E0407 12:38:20.358897 30068 util.go:91] /opt/app-root/etc/scl_enable: line 5: scl_source: No such file or directory
E0407 12:38:20.358920 30068 util.go:91] /opt/app-root/etc/scl_enable: line 6: scl_source: No such file or directory
---> Installing application source ...
---> Installing dependencies ...
Collecting gunicorn (from -r requirements.txt (line 1))
Downloading gunicorn-19.7.1-py2.py3-none-any.whl (111kB)
Installing collected packages: gunicorn
Successfully installed gunicorn-19.7.1

[INFO] Waiting for application container to start  ...


[INFO] Testing the image SCL enable

/opt/app-root/etc/scl_enable: line 5: scl_source: No such file or directory
/opt/app-root/etc/scl_enable: line 6: scl_source: No such file or directory
---> Running application from Python script (app.py) ...
LOADING MODULE app.py
[2018-04-07 10:38:28 +0000] [1] [INFO] Starting gunicorn 19.7.1
[2018-04-07 10:38:28 +0000] [1] [INFO] Listening at: http://0.0.0.0:8080 (1)
[2018-04-07 10:38:28 +0000] [1] [INFO] Using worker: sync
[2018-04-07 10:38:28 +0000] [17] [INFO] Booting worker with pid: 17

[INFO] Testing the image SCL enable

ERROR[/bin/bash -c node --version] Expected '^v6.', got '/opt/app-root/etc/scl_enable: line 5: scl_source: No such file or directory
/opt/app-root/etc/scl_enable: line 6: scl_source: No such file or directory
v8.9.4'

[INFO] TEST FAILED (1)


[INFO] Cleaning up the test application image

Untagged: f27/python3-testapp:latest
make: *** [common/common.mk:72: test] Error 1
```

--- a/2.7/s2i/bin/assemble
+++ b/2.7/s2i/bin/assemble
@@ -8,6 +8,14 @@ function should_collectstatic() {
   is_django_installed && [[ -z "$DISABLE_COLLECTSTATIC" ]]
 }
 
+function virtualenv_bin() {
+  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+    virtualenv-${PYTHON_VERSION} $1
+  else
+    virtualenv $1
+  fi
+}
+
 # Install pipenv to the separate virtualenv to isolate it
 # from system Python packages and packages in the main
 # virtualenv. Executable is simlinked into ~/.local/bin
@@ -17,7 +25,7 @@ function install_pipenv() {
   echo "---> Installing pipenv packaging tool ..."
   pip install -U virtualenv
   VENV_DIR=$HOME/.local/venvs/pipenv
-  virtualenv $VENV_DIR
+  virtualenv_bin "$VENV_DIR"
   $VENV_DIR/bin/pip --isolated install -U pipenv
   mkdir -p $HOME/.local/bin
   ln -s $VENV_DIR/bin/pipenv $HOME/.local/bin/pipenv

--- a/2.7/s2i/bin/assemble
+++ b/2.7/s2i/bin/assemble
@@ -9,7 +9,7 @@ function should_collectstatic() {
 }
 
 function virtualenv_bin() {
-  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+  if head "/etc/redhat-release" | grep -q "^Fedora"; then
     virtualenv-${PYTHON_VERSION} $1
   else
     virtualenv $1

--- a/3.4/s2i/bin/assemble
+++ b/3.4/s2i/bin/assemble
@@ -8,6 +8,14 @@ function should_collectstatic() {
   is_django_installed && [[ -z "$DISABLE_COLLECTSTATIC" ]]
 }
 
+function virtualenv_bin() {
+  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+    virtualenv-${PYTHON_VERSION} $1
+  else
+    virtualenv $1
+  fi
+}
+
 # Install pipenv to the separate virtualenv to isolate it
 # from system Python packages and packages in the main
 # virtualenv. Executable is simlinked into ~/.local/bin
@@ -17,7 +25,7 @@ function install_pipenv() {
   echo "---> Installing pipenv packaging tool ..."
   pip install -U virtualenv
   VENV_DIR=$HOME/.local/venvs/pipenv
-  virtualenv $VENV_DIR
+  virtualenv_bin $VENV_DIR
   $VENV_DIR/bin/pip --isolated install -U pipenv
   mkdir -p $HOME/.local/bin
   ln -s $VENV_DIR/bin/pipenv $HOME/.local/bin/pipenv

--- a/3.4/s2i/bin/assemble
+++ b/3.4/s2i/bin/assemble
@@ -9,7 +9,7 @@ function should_collectstatic() {
 }
 
 function virtualenv_bin() {
-  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+  if head "/etc/redhat-release" | grep -q "^Fedora"; then
     virtualenv-${PYTHON_VERSION} $1
   else
     virtualenv $1

--- a/3.5/s2i/bin/assemble
+++ b/3.5/s2i/bin/assemble
@@ -8,6 +8,14 @@ function should_collectstatic() {
   is_django_installed && [[ -z "$DISABLE_COLLECTSTATIC" ]]
 }
 
+function virtualenv_bin() {
+  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+    virtualenv-${PYTHON_VERSION} $1
+  else
+    virtualenv $1
+  fi
+}
+
 # Install pipenv to the separate virtualenv to isolate it
 # from system Python packages and packages in the main
 # virtualenv. Executable is simlinked into ~/.local/bin
@@ -17,7 +25,7 @@ function install_pipenv() {
   echo "---> Installing pipenv packaging tool ..."
   pip install -U virtualenv
   VENV_DIR=$HOME/.local/venvs/pipenv
-  virtualenv $VENV_DIR
+  virtualenv_bin $VENV_DIR
   $VENV_DIR/bin/pip --isolated install -U pipenv
   mkdir -p $HOME/.local/bin
   ln -s $VENV_DIR/bin/pipenv $HOME/.local/bin/pipenv

--- a/3.5/s2i/bin/assemble
+++ b/3.5/s2i/bin/assemble
@@ -9,7 +9,7 @@ function should_collectstatic() {
 }
 
 function virtualenv_bin() {
-  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+  if head "/etc/redhat-release" | grep -q "^Fedora"; then
     virtualenv-${PYTHON_VERSION} $1
   else
     virtualenv $1

--- a/3.6/s2i/bin/assemble
+++ b/3.6/s2i/bin/assemble
@@ -8,6 +8,14 @@ function should_collectstatic() {
   is_django_installed && [[ -z "$DISABLE_COLLECTSTATIC" ]]
 }
 
+function virtualenv_bin() {
+  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+    virtualenv-${PYTHON_VERSION} $1
+  else
+    virtualenv $1
+  fi
+}
+
 # Install pipenv to the separate virtualenv to isolate it
 # from system Python packages and packages in the main
 # virtualenv. Executable is simlinked into ~/.local/bin
@@ -16,7 +24,7 @@ function should_collectstatic() {
 function install_pipenv() {
   echo "---> Installing pipenv packaging tool ..."
   VENV_DIR=$HOME/.local/venvs/pipenv
-  virtualenv $VENV_DIR
+  virtualenv_bin $VENV_DIR
   $VENV_DIR/bin/pip --isolated install -U pipenv
   mkdir -p $HOME/.local/bin
   ln -s $VENV_DIR/bin/pipenv $HOME/.local/bin/pipenv

--- a/3.6/s2i/bin/assemble
+++ b/3.6/s2i/bin/assemble
@@ -9,7 +9,7 @@ function should_collectstatic() {
 }
 
 function virtualenv_bin() {
-  if [[ $(head /etc/redhat-release) =~ ^Fedora.* ]]; then
+  if head "/etc/redhat-release" | grep -q "^Fedora"; then
     virtualenv-${PYTHON_VERSION} $1
   else
     virtualenv $1

--- a/test/run
+++ b/test/run
@@ -165,9 +165,9 @@ test_application() {
 
   test_scl_usage "python --version" "Python $VERSION." "${cid_file}"
   check_result $?
-  test_scl_usage "node --version" "^v6." "${cid_file}"
+  test_scl_usage "node --version" "^v[0-9]*\.[0-9]*\.[0-9]*" "${cid_file}"
   check_result $?
-  test_scl_usage "npm --version" "^3." "${cid_file}"
+  test_scl_usage "npm --version" "^[0-9]*\.[0-9]*\.[0-9]*" "${cid_file}"
   check_result $?
   test_connection
   check_result $?

